pkg/ipam/crd.go

```diff
@@ -623,5 +624,7 @@ func (a *crdAllocator) Dump() (map[string]string, string) {
 
 // RestoreFinished marks the status of restoration as done
 func (a *crdAllocator) RestoreFinished() {
```

Comment:
The aim here is to restore all endpoints before triggering sync with upstream, so I think we should close `a.store.restoreFinished` only after the restoration of both `IPv6Allocator` and `IPv4Allocator` are finished. They are called from here:  
https://github.com/cilium/cilium/blob/90b256e93ff0b7545d269fef5146f47a45e19238/daemon/cmd/daemon.go#L491-L498

Reply:
@jaffcheng @Sen666666 Looking at the code, this seems like a potential fix that should go in but does not directly interact with this specific PR that just fixes a specific crash issue. I'd like to merge this PR as-is to fix the crash, then you can follow up to address these comments. Does that sound reasonable?

Reply:
Yes, let's address this comment in a follow-up PR.